sdlpal/sdlpal
# Makefile: Check dependency with gcc/clang's preprocessor option

As mentioned in #33 , Makefiles in current master lacks ability to detect actual dependency for rebuilding, which often makes problem in incremental debugging. Since all our makefiles uses gcc and clang is compatible of gcc's preprocessor options, let gcc deduce it should be a nice way.
Test steps:
1, `git checkout 17f2b88`, `make clean && make`
2, `git checkout b78537e`, `make`
With original makefiles, these will produce the mentioned bug. But with the makefiles in this branch( after every step's checkout, replace the makefiles, then start the latter make procedure ), the problem should not appear.

Current only unix/msys/mingw makefiles modified, if no other problem found, the same patch could also be apply to 3ds/wii/etc.
